core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionNetworkOutput.java

```diff
@@ -237,6 +250,24 @@ public boolean sendSync(Transmission transmission) {
         return true;
     }
 
+    private static void handleTemporaryExceptions(String message, Exception ex) {
+        //Handle first failure
+        if(!firstFailure.getAndSet(true)) {
```

Comment:
when will you reset firstFailure?

Reply:
first failure is only captured once throughout the life of the application. As per my understanding we dont reset firstFailure. @trask your thoughts on this?

Reply:
ya, we want special behavior for the first failure, to make sure it's logged right away. all remaining failures will only be logged every 5 min